neuralpredictors/layers/cores/conv2d.py

```diff
@@ -97,91 +102,124 @@ def __init__(
         """
 
         super().__init__()
-
         regularizer_config = (
             dict(padding=laplace_padding, kernel=input_kern)
             if input_regularizer == "GaussianLaplaceL2"
             else dict(padding=laplace_padding)
         )
         self._input_weights_regularizer = regularizers.__dict__[input_regularizer](**regularizer_config)
-
-        self.layers = layers
+        self.num_layers = layers
         self.gamma_input = gamma_input
         self.gamma_hidden = gamma_hidden
         self.input_channels = input_channels
         self.hidden_channels = hidden_channels
         self.skip = skip
+        self.stride = stride
         self.use_avg_reg = use_avg_reg
-
         if use_avg_reg:
             warnings.warn("The averaged value of regularizer will be used.", UserWarning)
-
-        self.features = nn.Sequential()
+        self.hidden_padding = hidden_padding
+        self.input_kern = input_kern
+        self.hidden_kern = hidden_kern
+        self.laplace_padding = laplace_padding
+        self.hidden_dilation = hidden_dilation
+        self.final_nonlinearity = final_nonlinearity
+        self.elu_xshift, self.elu_yshift = elu_shift
+        self.bias = bias
+        self.momentum = momentum
+        self.pad_input = pad_input
+        self.batch_norm = batch_norm
+        self.batch_norm_scale = batch_norm_scale
+        self.independent_bn_bias = independent_bn_bias
         if stack is None:
-            self.stack = range(self.layers)
+            self.stack = range(self.num_layers)
+        else:
+            self.stack = [*range(self.num_layers)[stack:]] if isinstance(stack, int) else stack
+        self.linear = linear
+
+        if depth_separable:
+            self.ConvLayer = DepthSeparableConv2d
+        elif attention_conv:
+            self.ConvLayer = self.AttentionConvWrapper
         else:
-            self.stack = [*range(self.layers)[stack:]] if isinstance(stack, int) else stack
+            self.ConvLayer = nn.Conv2d
```

Comment:
I would have preferred to have the `ConvLayer` as an argument to the `__init__` or a class attribute but I guess that is not possible without changing the interface of the class.